src/preprocessing/filtering.jl

```diff
@@ -1,88 +1,145 @@
+                s_dim2 = size(s)[2]
+            else
+                s_dim2 = 1
+            end
+			s = vcat(s,zeros(tau*2,s_dim2))
+            
+			
+			# filter
+			s = filt(fobj,s)
+			
+			# fix grpdelay
+			s = s[tau+1:end-tau,:]
+			
+		end
+		return s
+	end
+	function filterdelay(fobj::Vector) 
+        return (length(fobj)-1)÷2
+    end
+    function filterdelay(fobj::ZeroPoleGain)
+        # Todo: This is unsatisfactory, but maybe necessary?
+        error("Butterworth has non-linear phase, use filtfilt=true to compensate delay")
+    end
+
+	function  default_fir_filterorder(responsetype,samplingrate)
+		# filter settings are the same as firfilt eeglab plugin (Andreas Widmann) and MNE Python. 
+		# filter order is set to 3.3 times the reciprocal of the shortest transition band 
+		# transition band is set to either
+		# min(max(l_freq * 0.25, 2), l_freq)
+		# or 
+		# min(max(h_freq * 0.25, 2.), nyquist - h_freq)
+		# 
+		# That is, 0.25 times the frequency, but maximally 2Hz
+		%
+		
+			transwidthratio = 0.25 # magic number from firfilt eeglab plugin
+			fNyquist = samplingrate ./ 2
+			cutOff = responsetype.w * samplingrate
+			# what is the maximal filter width we can have
+			if typeof(responsetype) <: Highpass
+				maxDf = cutOff
+			
+				df = minimum([maximum([maxDf * transwidthratio, 2]), maxDf]);
+
+			elseif typeof(responsetype) <: Lowpass
+				#for lowpass we have to look back from nyquist
+				maxDf = fNyquist - cutOff
+				df = minimum([maximum([cutOff * transwidthratio, 2]), maxDf]);
+
+			end
+			
+			filterorder = 3.3 ./ (df ./ samplingrate) 
+			filterorder = Int(filterorder ÷ 2 * 2) # we need even filter order
+		return filterorder
+	end
 
-Simply a wrapper for the DSP.jl functions
+"""
+    filter_highpass(filter_highpass(obj::NeuroimagingMeasurement ,cutOff;kwargs...)
+
+High pass filter, group delay corrected. Modality specific defaults are applied
 
 # Arguments
 
-* `signals`: Signal data in the format samples x channels
+* `obj`: NeuroimagingMeasurement
 * `cutOff`: Cut off frequency in Hz
-* `fs`: Sampling rate
-* `order`: Filter orde
+* `kwargs...` passed on to filter
 
 # Returns
 
-* filtered signal
-* filter used on signal
+* filtered NeuroimagingMeasurement
 """
-function highpass_filter(
-    signals::Array{T},
-    cutOff::Number,
-    fs::Number,
-    order::Int,
-) where {T<:AbstractFloat}
-    @debug("Highpass filtering $(size(signals)[end]) channels.  Pass band > $(cutOff) Hz")
-    Wn = cutOff / (fs / 2)
-    highpass_filter(signals, Wn, order)
-end
-
-
-function highpass_filter(signals::Array{T}, Wn::Number, order::Int) where {T<:AbstractFloat}
-    @debug("Filter order = $order, Wn = $Wn")
-    f = digitalfilter(Highpass(Wn), Butterworth(order))
-    signals = filtfilt(f, signals)
-    return signals, f
+function filter_highpass(obj::NeuroimagingMeasurement ,cutOff;kwargs...)
+    responsetype = Highpass(cutOff,fs=samplingrate(obj))
+    return filter(obj,responsetype;kwargs...)

```

Comment:
Functions operating on types should go in `/src/types/..` So this code would belong in `src/types/NeuroimagingMeasurement.jl` or `src/types/NeuroimagingMeasurement/preprocessing.jl`.